index.js

```diff
@@ -394,20 +428,14 @@ Color.prototype.setValues = function (space, vals) {
 
 	// cap values of the space prior converting all values
 	for (i = 0; i < space.length; i++) {
-		capped = Math.max(0, Math.min(maxes[space][i], this.values[space][i]));
-		this.values[space][i] = Math.round(capped);
+		capped = Math.max(0, Math.min(maxes[space][i], values[space][i]));
+		values[space][i] = Math.round(capped);
 	}
 
 	// convert to all the other color spaces
 	for (var sname in spaces) {
 		if (sname !== space) {
-			this.values[sname] = convert[space][sname](this.values[space]);
-		}
-
-		// cap values
-		for (i = 0; i < sname.length; i++) {
```

Comment:
I explained why in the [associated commit](https://github.com/chartjs/chartjs-color/pull/1/commits/4fb3e1068e7ebde7d8ed3c3787d3426d23b5cb7f) ;) We already cap source values before doing all conversions, and I'm pretty sure that conversion methods return valid values if source values are valid. The extra `Math.round()` was useless since conversion methods already return rounded values. These changes don't break unit tests, so should be ok I guess :)